### PR TITLE
docs: add index-comparison sheet's demo

### PR DESCRIPTION
**Pull request by yangzhanmei**
### 👀 PR includes

🔧 Chore

- [ ] Test case
- [x] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description
docs: add index-comparison sheet's demo
### 🖼️ Screenshot

Kapture 2021-11-05 at 18 26 41
| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### CODE REVIEW

File: s2-site/examples/data/index-comparison.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
+[
+  {
```
Comment (xingwanying) [inline s2-site/examples/data/index-comparison.json:2]: 这个demo是先占位？

**Review by xingwanying — COMMENTED** [review]
个人建议：没必要加在_tests_里，直接挪到场景demo里就好

File: s2-site/examples/data/index-comparison.json (hunk 1 above)
Comment (yangzhanmei) [reply s2-site/examples/data/index-comparison.json:2]: 对的

**Review by xingwanying — APPROVED** [review]
TODO：
image

- [ ] 截图替换
- [ ] 设计指南添加

**Pull request merged**